Feature/multi user signing frontend

dhelmert opened the pull request:
### Reason for this PR
- The operation management is needed for Multi-User-Signing, to handle access rights, operation logic and caching

### Changes in this PR
- Add operation-management.yaml with endpoints for fetching operations with various filters

dhelmert committed:
Add operation-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Merge branch 'develop' into feature/multi_user_signing_frontend

dhelmert committed:
Fix description of delete

dhelmert committed:
Update query parameters to conform to new operation functionality

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

bastihav commented on operation-management.yaml:
```diff
@@ -0,0 +1,301 @@
+openapi: "3.0.0"
+info:
+  description: "This is the Operation API for UC4."
+  version: "0.12.1"
```
```suggestion
  version: "0.15.1"
```

bastihav commented on operation-management.yaml:
```diff
@@ -0,0 +1,301 @@
+  title: "UC4"
+
+servers:
+  - url: https://uc4.cs.upb.de/api/operation-management
```
```suggestion
  - url: https://uc4.cs.upb.de/api/production/operation-management
```

bastihav commented on operation-management.yaml:
```diff
@@ -0,0 +1,301 @@
+    description: "Find out more"
+    url: "https://github.com/upb-uc4"
+
+paths:
```
hl version missing

dhelmert replied:
Has been added as of now

bastihav commented on operation-management.yaml:
```diff
@@ -0,0 +1,301 @@
+          type: 'boolean'
+      - name: "selfActionRequired"
+        in: "query"
+        description: "Flag that determines, if only operations that you can approve/deny are fetched"
```
```suggestion
        description: "Flag that determines, if only operations that you can approve/reject are fetched"
```

bastihav commented on operation-management.yaml:
```diff
@@ -0,0 +1,301 @@
+          type: 'boolean'
+      - name: "states"
+        in: "query"
+        description: "List of states to filter by (PENDING, FINISHED, or REJECTED)"
```
```suggestion
        description: "Comma-seperated list of operation states to filter by (PENDING, FINISHED, or REJECTED)"
```

bastihav commented on operation-management.yaml:
```diff
@@ -0,0 +1,301 @@
+      tags:
+      - "Operation Management"
+      summary: "Find Operation by OperationId"
+      description: "Find course by OperationId"
```
```suggestion
      description: "Find operation by OperationId"
```

dhelmert committed:
Update operation-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Update operation-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Update operation-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Update operation-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Update operation-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Merge branch 'develop' into feature/multi_user_signing_frontend

dhelmert committed:
Apply new version handling to operation api

dhelmert committed:
Add different api urls to operation-management.yaml

dhelmert committed:
Rename Proposal Objects in matriculation
Add Reject Proposal

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Add last-modified and initiator to operation object

Co-authored-by: tobias-wiese <tobiaswiese@online.de>
Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Merge branch 'develop' into feature/multi_user_signing_frontend

dhelmert committed:
Add timestamps to operation-management

bastihav reviewed (changes requested):
examples for matriculation payload are wrong due to approve/reject restructuring

dhelmert committed:
Add proposal logic to operation management
Move Rejection from matriculation to operation
Refactor some matriculation endpoints
Remove deprecated matriculation endpoint

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Rename cacheOnly to watchlistOnly, since there is no cache on lagom anymore

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Rename reject message, to be more consistent to our other apis

dhelmert committed:
Add new endpoint for fetching username of enrollmentId
Add username back to Matriculation getProposal

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert commented:
> 
> 
> examples for matriculation payload are wrong due to approve/reject restructuring

This is a minimal working api, the new changes, for which these examples are wreong will be coming in another branch
